CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -2752,13 +2752,9 @@ public function processConfirm(
           }
           $result = $payment->doPayment($paymentParams);
           $form->_params = array_merge($form->_params, $result);
-          $form->assign('trxn_id', CRM_Utils_Array::value('trxn_id', $result));
-          if (!empty($result['trxn_id'])) {
-            $contribution->trxn_id = $result['trxn_id'];
-          }
-          if (!empty($result['payment_status_id'])) {
-            $contribution->payment_status_id = $result['payment_status_id'];
-          }
+          $form->assign('trxn_id', $result['trxn_id'] ?? '');
+          $contribution->trxn_id = $result['trxn_id'] ?? $contribution->trxn_id ?? '';
+          $contribution->payment_status_id = $result['payment_status_id'];
```

Comment:
Any payment processor that does not return `payment_status_id` is going to be broken in multiple ways because it is the only way to check if payment was successful..